File: java/Storage.java
import java.util.ArrayList;

public class Storage extends Komponen implements AutoCloseable {
    private int kapasitasGB;
    private ArrayList<HardDrive> harddrives;
    private ArrayList<SSD> ssds;

    public Storage(String merk, String nama, int kapasitasGB) {
        super(merk, nama);
        this.kapasitasGB = kapasitasGB;
        this.harddrives = new ArrayList<>();
        this.ssds = new ArrayList<>();
    }

    @Override
    public void close() {
        // Hapus semua harddrives dan SSDs
        for (HardDrive hdd : harddrives) {
            hdd = null;
        }
        for (SSD ssd : ssds) {
            ssd = null;
        }
    }

    // Getter
    public int getKapasitasGB() {
        return kapasitasGB;
    }

    // Setter
    public void setKapasitasGB(int kapasitasGB) {
        this.kapasitasGB = kapasitasGB;
    }

    // Tambah HDD dan SSD
    public void addHarddrive(HardDrive hdd) {
        harddrives.add(hdd);
    }

    public void addSsd(SSD ssd) {
        ssds.add(ssd);
    }

    // Method info
    public String info() {
        StringBuilder result = new StringBuilder("Storage: " + getNama() + " " + kapasitasGB + "GB (Merk: " + getMerk() + ")\n");
        result.append("  Harddrives:\n");
        for (HardDrive hdd : harddrives) {
            result.append("    - ").append(hdd.info()).append("\n");
        }
        result.append("  SSDs:\n");
        for (SSD ssd : ssds) {
            result.append("    - ").append(ssd.info()).append("\n");
        }
        return result.toString();
    }
}

File: Python/Storage.py
from Komponen import Komponen
from Hardrive import Harddrive
from SSD import SSD

class Storage(Komponen):
    def __init__(self, merk="", nama="", kapasitasGB=0):
        super().__init__(merk, nama)
        self.kapasitasGB = kapasitasGB
        self.harddrives = []
        self.ssds = []

    def __del__(self):
        # Hapus semua harddrives dan SSDs
        for hdd in self.harddrives:
            del hdd
        for ssd in self.ssds:
            del ssd

    # Getter
    def get_kapasitasGB(self):
        return self.kapasitasGB

    # Setter
    def set_kapasitasGB(self, kapasitasGB):
        self.kapasitasGB = kapasitasGB

    # Tambah HDD dan SSD
    def add_harddrive(self, hdd):
        self.harddrives.append(hdd)

    def add_ssd(self, ssd):
        self.ssds.append(ssd)

    # Method info
    def info(self):
        result = f"Storage: {self.get_nama()} {self.kapasitasGB}GB (Merk: {self.get_merk()})\n"
        result += "  Harddrives:\n"
        for hdd in self.harddrives:
            result += f"    - {hdd.info()}\n"
        result += "  SSDs:\n"
        for ssd in self.ssds:
            result += f"    - {ssd.info()}\n"
        return result